src/main/java/io/mbab/sda/groupproject/service/CountryService.java

```diff
@@ -0,0 +1,52 @@
+package io.mbab.sda.groupproject.service;
+
+import io.mbab.sda.groupproject.entity.Country;
+import io.mbab.sda.groupproject.repository.CountryRepository;
+import lombok.RequiredArgsConstructor;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+
+@RequiredArgsConstructor
+public class CountryService implements CrudService<Country, Integer> {
+
+  private final CountryRepository countryRepository;
+  private final EntityManager em;
+
+  public Country save(Country country) {
+    try {
+      em.getTransaction().begin();
+      if (country.getId() == null) {
+        country = countryRepository.create(country);
```

Comment:
jeżeli nie ma ID to persystujesz, po czym znów persystujesz? W serwisie EntityManager ma nam służyć tylko dla bloku transakcji